lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1418 @@
+              callback(err);
+            });
+        },
+        (result) => {
+          if (result instanceof Error) {
+            throw result;
+          } else {
+            if (result === null) {
+              this.pumpSession.radioChannel = 0;
+              reject(new Error('Please make sure that the pump is paired with this ' +
+                'Contour Next Link 2.4 and that the pump is in range'));
+            }
+            resolve();
+          }
+        },
+      );
+    });
+  }
+
+  toggleHighSpeedMode(mode) {
+    return new HighSpeedModeCommand(this.pumpSession, mode).send(this.hidDevice);
+  }
+
+  getPumpTime() {
+    return new PumpTimeCommand(this.pumpSession).send(this.hidDevice);
+  }
+
+  getPumpStatus() {
+    return new PumpStatusCommand(this.pumpSession).send(this.hidDevice);
+  }
+
+  getBolusWizardSettings() {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      const settings = {};
+      new BolusWizardBGTargetsCommand(this.pumpSession).send(this.hidDevice)
+        .then((response) => {
+          settings.bgTarget = response.targets;
+        })
+        .then(() => new BolusWizardCarbRatiosCommand(this.pumpSession).send(this.hidDevice))
+        .then((response) => {
+          settings.carbRatio = response.ratios;
+        })
+        .then(() => new BolusWizardSensitivityFactorsCommand(this.pumpSession).send(this.hidDevice))
+        .then((response) => {
+          settings.insulinSensitivity = response.factors;
+        })
+        .then(() => resolve(settings));
+    });
+  }
+
+  getDeviceCharacteristics() {
+    return new DeviceCharacteristicsCommand(this.pumpSession).send(this.hidDevice);
+  }
+
+  getDeviceString() {
+    return new DeviceStringCommand(this.pumpSession).send(this.hidDevice);
+  }
+
+  readBasalPatterns() {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      async.timesSeries(8,
+        (n, next) => {
+          new ReadBasalPatternCommand(this.pumpSession, n + 1).send(this.hidDevice)
+            .then((response) => {
+              next(null, response.schedule);
+            })
+            .catch(err => next(err, null));
+        },
+        (err, results) => {
+          if (err) {
+            reject(err);
+          } else {
+            const basalSchedules = {};
+            for (let i = 0; i < results.length; i++) {
+              const schedule = results[i];
+              // Only include patterns with schedules in them
+              if (schedule.length > 0) {
+                basalSchedules[MM600SeriesDriver.PATTERN_NAME[i]] = results[i];
+              }
+            }
+            resolve(basalSchedules);
+          }
+        });
+    });
+  }
+}
+
+module.exports = (config) => {
+  const cfg = _.clone(config);
+  const driver = new MM600SeriesDriver(cfg.deviceComms);
+
+  // REVIEW - discuss http://eslint.org/docs/rules/no-param-reassign...
```

Comment:
It's not worth trying to fix existing drivers, but I'm wondering if cloning objects is a good best practice in this and any new drivers. Thoughts @jebeck?